src/Orleans/Utils/IAsymmetricComparer.cs

```diff
@@ -0,0 +1,16 @@
+﻿
+namespace Orleans.Runtime
+{
+    public interface IAsymmetricComparer<in T1, in T2>
```

Comment:
@gabikliot To use the pooled cache for other providers like kafka, the data structures in the cache need to be defined by the provider.  This class is used to compare different types in the pooled cache, so provider developers will need to implement it to use the pooled cache.  This is why it is public and not internal.  I added it to the utils because it seemed more generic than the other interfaces.

I'm perfectly willing to change this if we can find a better way.  Can you suggest a better approach?

Some options:
- Move the comparer behavior back into the data adapter.  This complicates the data adapter extensibility point in the EventHubProvider, but I suspect this is solvable.
- Make pooled cache EventHub only.  This reduces the complexity of the data adapter (still does not remove it), but means we would still not have a cache available for rewindable stream provider developers.

I'm vary open to alternatives, but of the options I can think of, the costs outway the advantages.